qiskit/compiler/assembler.py

```diff
@@ -34,13 +34,47 @@ def assemble_circuits(circuits, qobj_id=None, qobj_header=None, run_config=None)
         qobj_header (QobjHeader): header to pass to the results
         run_config (RunConfig): configuration of the runtime environment
 
+    Raises:
+        QiskitError: if run_config parameters are not compatible with circuit parameters
+
     Returns:
         QasmQobj: the Qobj to be run on the backends
     """
     qobj_config = QasmQobjConfig()
     if run_config:
         qobj_config = QasmQobjConfig(**run_config.to_dict())
 
+    # Bind parameters (if specified) or circuits are parameterized
+    parameter_binds = getattr(run_config, 'parameter_binds', [])
+    if parameter_binds or \
+       any(circuit.parameters for circuit in circuits):
+
+        all_bind_parameters = [bind.keys()
+                               for bind in parameter_binds]  # pylint: disable=not-an-iterable
+        all_circuit_parameters = [circuit.parameters for circuit in circuits]
+
+        # Collect set of all unique parameters across all circuits and binds
+        unique_parameters = set(param
+                                for param_list in all_bind_parameters + all_circuit_parameters
+                                for param in param_list)
+
+        # Check that all parameters are common to all circuits and binds
+        if not all_bind_parameters \
+           or not all_circuit_parameters \
+           or any(unique_parameters != bind_params for bind_params in all_bind_parameters) \
+           or any(unique_parameters != parameters for parameters in all_circuit_parameters):
+            raise QiskitError(
+                ('Mismatch between run_config.parameter_binds and all circuit parameters. ' +
+                 'Parameter binds: {} ' +
+                 'Circuit parameters: {}').format(all_bind_parameters, all_circuit_parameters))
+
+        circuits = [circuit.bind_parameters(binds)
+                    for circuit in circuits
+                    for binds in parameter_binds]  # pylint: disable=not-an-iterable
+
+        # All parameters have been expanded and bound, so remove from run_config
+        run_config.parameter_binds = []
```

Comment:
because line 45 has already copied `run_config` to `qobj_config`, the parameter binding dicts end up leaking into the final Qobj config. It won't affect the run, but probably better to not pollute the config with useless info.

Reply:
👍 Fixed in 77925cea87dacf62354d06c0bbf6a7c20ef9f59c.